cmd/vm/switch_linux.go

```diff
@@ -140,6 +149,34 @@ func run(ctx context.Context, cancel context.CancelFunc, connFile io.ReadWriteCl
 	return <-errCh
 }
 
+func forwardLoopback(subnet string) error {
+	ip, _, err := net.ParseCIDR(subnet)
+	if err != nil {
+		return err
+	}
+	tapDeviceIP := config.TapDeviceIP(ip.To4())
+	rules := map[string][]string{
+		"PREROUTING":  {"-t", "nat", "-A", "PREROUTING", "-d", tapDeviceIP, "-j", "DNAT", "--to-destination", "127.0.0.1"},
```

Comment:
Does that mean if I have two containers, one listening on `127.0.0.1:80`, and one listening on `192.168.127.2:80` (the tap device IP, in case I got that wrong), we'll _always_ send traffic to the former? In fact, if we only have the latter, it'll never receive traffic despite nothing is listening on the former?

(The only scenario I can think of where this matters is actually two containers, `127.0.0.1:80` plus `0.0.0.0:80`.)

Reply:
I don't think this is a real scenario and the userland proxy (`host-switch.exe`) does have a guard for this and will error out. Consider the following:
1) A user uses port `7000` to deploy a container attached to `127.0.0.1`
e.g:
```
docker run -d -p  127.0.0.1:7000:80 nginx
```
and now you will not be able to deploy a second container on the same port `7000` attached to `0.0.0.0`, the `host-switch.exe` will error out letting the user know that the port is occupied. 

2) A user will attempt to use port `7000` to deploy a container attached to `0.0.0.0`, but the user is not able to re-use that port on `127.0.0.1` since the port is already occupied using `INADDR_ANY` and that includes all local network interfaces (including loopback).
e.g:
```
docker run -d -p  7000:80 nginx
```
3) A user will attempt something clever where they will bind directly to an internal IP address.
e.g:
```
docker run -d -p  192.168.127.2:7000:80 nginx
```
While this a legitimate use case, however, the above port is being forwarded on the host as `127.0.0.1:7000`, 
therefore any attempts to re-use port `7000` on `0.0.0.0` or `127.0.0.1` will prevent the user from successfully running the command and will be prevented by an error from `host-switch.exe`